keras/src/losses/losses.py

```diff
@@ -23,7 +23,6 @@ def __init__(
         self._fn_kwargs = kwargs
 
     def call(self, y_true, y_pred):
-        y_true, y_pred = squeeze_or_expand_to_same_rank(y_true, y_pred)
```

Comment:
Remove this is likely to break some cases -- we should make sure we keep per-output squeezing/expansion

Reply:
So far only one _minor_ unit test was affected cf your [comment](https://github.com/keras-team/keras/pull/20358#discussion_r1803589503).
I'll put it back.

Reply:
Put it back.